packages/adblocker/src/filters/cosmetic.ts

```diff
@@ -585,6 +642,20 @@ export default class CosmeticFilter implements IFilter {
       return false;
     }
 
+    if (
+      this.parentDomains !== undefined &&
+      this.parentDomains.match(
+        hostname.length === 0
+          ? EMPTY_UINT32_ARRAY
+          : getHostnameHashesFromLabelsBackward(hostname, domain),
+        hostname.length === 0
+          ? EMPTY_UINT32_ARRAY
+          : getEntityHashesFromLabelsBackward(hostname, domain),
+      )
+    ) {
+      return true;
+    }
```

Comment:
This is where the detailed matching happens. If a filter has `parentDomains` value, then we will try to match `parentDomains` with the given hostname (which would be both current and parent hostnames from higher APIs).